securevault/src/main/java/org/wso2/carbon/securevault/internal/SecureVaultComponent.java

```diff
@@ -0,0 +1,143 @@
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.securevault.internal;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.securevault.MasterKeyReader;
+import org.wso2.carbon.securevault.SecretRepository;
+import org.wso2.carbon.securevault.SecureVault;
+import org.wso2.carbon.securevault.SecureVaultInitializer;
+
+/**
+ * This service component acts as a RequiredCapabilityListener for all the ${@link SecretRepository}s and
+ * ${@link MasterKeyReader}s. This component will receive all the ${@link SecretRepository} and
+ * ${@link MasterKeyReader} services registrations, but it will only keep references for the services that are
+ * configured in the secure-vault.yaml. Once all the services are available, this component will initialize the
+ * corresponding ${@link SecretRepository} and ${@link MasterKeyReader} and call the ${@link SecretRepository}
+ * to load the secrets. Once the ${@link SecretRepository} is ready, this component will  register the
+ * SecureVault OSGi service, which can then be used by other components for encryption and decryption.
+ *
+ * @since 5.2.0
+ */
+@Component(
+        name = "org.wso2.carbon.securevault.internal.SecureVaultComponent",
+        immediate = true
+)
+public class SecureVaultComponent {
+
+    private static final Logger logger = LoggerFactory.getLogger(SecureVaultComponent.class);
+
+    public SecureVaultComponent() {
+        SecureVaultInitializer.getInstance().initFromSecureVaultYAML();
+    }
+
+    @Activate
+    public void activate(BundleContext bundleContext) {
+        logger.debug("Activating SecureVaultComponent");
+    }
+
+    @Deactivate
+    public void deactivate(BundleContext bundleContext) {
+        SecureVaultDataHolder.getInstance().setBundleContext(null);
+        logger.debug("Deactivating SecureVaultComponent");
+    }
+
+    @Reference(
+            name = "secure.vault.secret.repository",
+            service = SecretRepository.class,
+            cardinality = ReferenceCardinality.AT_LEAST_ONE,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unRegisterSecretRepository"
+    )
+    protected void registerSecretRepository(SecretRepository secretRepository) {
+        if (secretRepository.getClass().getName().equals(SecureVaultInitializer.getInstance()
+                .getSecretRepositoryType())) {
+            logger.debug("Registering secret repository : {}", SecureVaultInitializer.getInstance()
+                    .getSecretRepositoryType());
```

Comment:
We can move whole SecureVaultInitializer.getInstance().getSecretRepositoryType() to the second line since it already takes two lines and it looks better.